eng/common/scripts/Helpers/AzSdkTool-Helpers.ps1

```diff
@@ -0,0 +1,192 @@
+Set-StrictMode -Version 4
+
+function Get-SystemArchitecture {
+    $unameOutput = uname -m
+    switch ($unameOutput) {
+        "x86_64" { return "X86_64" }
+        "aarch64" { return "ARM64" }
+        "arm64" { return "ARM64" }
+        default { throw "Unable to determine system architecture. uname -m returned $unameOutput." }
+    }
+}
+
+function Get-Package-Meta(
+    [Parameter(mandatory = $true)]
+    $FileName,
+    [Parameter(mandatory = $true)]
+    $Package
+) {
+    $ErrorActionPreferenceDefault = $ErrorActionPreference
+    $ErrorActionPreference = "Stop"
+
+    $AVAILABLE_BINARIES = @{
+        "Windows" = @{
+            "AMD64" = @{
+                "system"     = "Windows"
+                "machine"    = "AMD64"
+                "file_name"  = "$FileName-standalone-win-x64.zip"
+                "executable" = "$Package.exe"
+            }
+        }
+        "Linux"   = @{
+            "X86_64" = @{
+                "system"     = "Linux"
+                "machine"    = "X86_64"
+                "file_name"  = "$FileName-standalone-linux-x64.tar.gz"
+                "executable" = "$Package"
+            }
+            "ARM64"  = @{
+                "system"     = "Linux"
+                "machine"    = "ARM64"
+                "file_name"  = "$FileName-standalone-linux-arm64.tar.gz"
+                "executable" = "$Package"
+            }
+        }
+        "Darwin"  = @{
+            "X86_64" = @{
+                "system"     = "Darwin"
+                "machine"    = "X86_64"
+                "file_name"  = "$FileName-standalone-osx-x64.zip"
+                "executable" = "$Package"
+            }
+            "ARM64"  = @{
+                "system"     = "Darwin"
+                "machine"    = "ARM64"
+                "file_name"  = "$FileName-standalone-osx-arm64.zip"
+                "executable" = "$Package"
+            }
+        }
+    }
+
+    $os = "unknown"
+    $machine = Get-SystemArchitecture
+
+    if ($IsWindows) {
+        $os = "Windows"
+        # we only support x64 on windows, if that doesn't work the platform is unsupported
+        $machine = "AMD64"
+    }
+    elseif ($IsLinux) {
+        $os = "Linux"
+    }
+    elseif ($IsMacOS) {
+        $os = "Darwin"
+    }
+
+    $ErrorActionPreference = $ErrorActionPreferenceDefault
+
+    return $AVAILABLE_BINARIES[$os][$machine]
+}
+
+function Cleanup-Directory ($path) {
+    if (Test-Path -Path $path) {
+        Remove-Item -Path $path -Recurse -Force
+    }
+    New-Item -ItemType Directory -Path $path -Force
+}
+
+function Is-Work-Necessary (
```

Comment:
The name of this function is maybe a little generic. I suggest something like `Is-Version-Downloaded`